clang-tools-extra/clang-tidy/bugprone/MethodHidingCheck.cpp

```diff
@@ -0,0 +1,141 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "MethodHidingCheck.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+#include <stack>
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::bugprone {
+
+namespace {
+
+bool sameBasicType(ParmVarDecl const *Lhs, ParmVarDecl const *Rhs) {
```

Comment:
```suggestion
bool sameBasicType(const ParmVarDecl *Lhs, const ParmVarDecl *Rhs) {
```
And the same everywhere (that is the style we use in `clang-tidy`).
I'm really confused that `clang-format` don't flag it. Is both formats are allowed in LLVM..? I'd make some research.

Reply:
So, the default value in LLVM is `QualifierAlignment: Leave` but I think de-facto we use `QualifierAlignment: Left` in `clang-tidy`. I'll try to enable it, and you may re-run `clang-format` over your code when it's done.

Reply:
Will do - let me know when your clang-format change lands in main

Reply:
Actually no, I'll fix this by hand, it's really not that hard